src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,800 @@
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        return routeRO.wrap(buffer, index, index + length);
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer throttle)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long streamId = begin.streamId();
+
+        MessageConsumer newStream = null;
+
+        if ((streamId & 0x0000_0000_0000_0001L) != 0L)
+        {
+            newStream = newInitialStream(begin, throttle);
+        }
+        else
+        {
+            newStream = newReplyStream(begin, throttle);
+        }
+        return newStream;
+    }
+
+    private MessageConsumer newInitialStream(
+        final BeginFW begin,
+        final MessageConsumer sender)
+    {
+        final long routeId = begin.routeId();
+        final long initialId = begin.streamId();
+        final long replyId = supplyReplyId.applyAsLong(initialId);
+
+        final MessagePredicate filter = (t, b, o, l) ->
+        {
+            return true;
+        };
+
+        final RouteFW route = router.resolve(routeId, begin.authorization(), filter, this::wrapRoute);
+        MessageConsumer newStream = null;
+
+        if (route != null)
+        {
+            final SocksServer connection = new SocksServer(sender, routeId, initialId, replyId);
+            newStream = connection::onNetwork;
+        }
+        return newStream;
+    }
+
+    private MessageConsumer newReplyStream(
+        final BeginFW begin,
+        final MessageConsumer sender)
+    {
+        final long replyId = begin.streamId();
+        final SocksServerStream connect = correlations.remove(replyId);
+
+        MessageConsumer newStream = null;
+        if (connect != null)
+        {
+            newStream = connect::onApplication;
+        }
+        return newStream;
+    }
+
+    private final class SocksServer
+    {
+        private final MessageConsumer network;
+        private final long routeId;
+        private final long initialId;
+        private final long replyId;
+        private long authorization;
+
+        private int initialBudget;
+        private int initialPadding;
+        private int replyBudget;
+        private int replyPadding;
+
+        private long decodeTraceId;
+        private DecoderState decodeState;
+        private int bufferSlot = BufferPool.NO_SLOT;
+        private int bufferSlotOffset;
+
+        private SocksServer(
+            MessageConsumer network,
+            long routeId,
+            long initialId,
+            long replyId)
+        {
+            this.network = network;
+            this.routeId = routeId;
+            this.initialId = initialId;
+            this.replyId = replyId;
+            this.decodeState = this::decodeCommandType;
+        }
+
+        private int decodeCommandType(
```

Comment:
Rename `decodeCommandType` to `decodeCommand` and remove `socksCommandType` parameter.